File: src/test/java/easy/StringIntegerIsBalancedTest.java
package easy;

import static org.junit.jupiter.api.Assertions.assertEquals;

import java.util.stream.Stream;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.Arguments;
import org.junit.jupiter.params.provider.MethodSource;

public class StringIntegerIsBalancedTest {
  StringIntegerIsBalanced instance = new StringIntegerIsBalanced();

  @ParameterizedTest
  @MethodSource("dataStringIntegerIsBalanced")
  public void testStringIntegerIsBalanced(String num, boolean expected) {

    boolean result = instance.solve(num);

    assertEquals(expected, result);
  }

  public static Stream<Arguments> dataStringIntegerIsBalanced() {
    return Stream.of(
        Arguments.of("12", false),
        Arguments.of("11", true),
        Arguments.of("1234", false),
        Arguments.of("24123", true),
        // Edge / special cases
        Arguments.of("00", true),
        Arguments.of("10", false),
        Arguments.of("1010", false),
        Arguments.of("909", false),
        Arguments.of("1234567890", false),
        Arguments.of("111111", true),
        Arguments.of("123321", true),
        Arguments.of("99999999", true),
        Arguments.of("5005", true),
        Arguments.of("1234567", false),
        // Optional out-of-constraint test
        Arguments.of("1", false));
  }
}

File: easy/string_integer_is_balanced_test.go
package easy

import (
	"testing"
)

func TestStringIntegerIsBalanced(t *testing.T) {
	cases := []struct {
		num      string
		expected bool
	}{
		{"12", false},
		{"11", true},
		{"1234", false},
		{"24123", true},
		{"00", true},
		{"10", false},
		{"1010", false},
		{"909", false},
		{"1234567890", false},
		{"111111", true},
		{"123321", true},
		{"99999999", true},
		{"5005", true},
		{"1234567", false},
		{"1", false},
	}
	for _, c := range cases {
		t.Run("", func(t *testing.T) {
			result := StringIntegerIsBalanced(c.num)
			if result != c.expected {
				t.Errorf("StringIntegerIsBalanced(%q) == %v, expected %v", c.num, result, c.expected)
			}

		})

	}
}
